os/os-mac.h

```diff
@@ -117,3 +122,60 @@ static inline bool os_cpu_has(cpu_features feature)
 }
 
 #endif
+
+#if defined (__APPLE__)
+/*
+ * discard_pages should not be run within Rosetta. Native arm64 or x86 only. 
+ */
+
+#define MMAP_CHUNK_SIZE		(1024 * 1024 * 1024)
+
+static inline int discard_pages(int fd, off_t offset, off_t size)
+{
+	caddr_t *addr;
+	uint64_t chunk_size = MMAP_CHUNK_SIZE;
+
+	if (fsync(fd) < 0) {
+                int __err = errno; 
+		fprintf(stderr, "Cannot fsync file\n");
```

Comment:
I think we would be better off using `log_err()`. Doing a quick grep there isn't a clear standard for how log_err's string should start but perhaps  a prefix of "discard_pages: " would make it clear where the error bubbled up from?